app/utils/vcs.js

```diff
@@ -0,0 +1,59 @@
+import config from 'travis/config/environment';
+
+const { providers } = config;
+
+const vcsId = (vcsType) => {
+  const availableProvidersRegex = Object.keys(providers).join('|');
+  const match = (vcsType || 'github').match(new RegExp(availableProvidersRegex, 'i'));
+
+  if (!match[0]) {
+    throw new Error(`Invalid VCS Type "${vcsType}"`);
+  }
+  return match[0].toLowerCase();
```

Comment:
Is it necessary to run the RegExp search in `vcsType`? It looks to me like it should just be a provider key, shouldn't it? Wouldn't 
```js
const vcsId = (vcsType = 'github') => {
  vcsType = vcsType.toLowerCase();
  assert(`Invalid VCS Type "${vcsType}"`, Object.keys(providers).includes(vcsType));
  return vcsType;
}
```
work here?

Reply:
Would it makes sense to make a dictionary of possible values and map them to `vcsId`s explicitly? Or even put to config as, let's say, aliases. My concern here is that current approach could potentially cause unpredictable and hard-to-track bugs, like if someone eventually would add some other `vcsType` without mentioning a provider - in this case UI will just silently brake. What do you think?

Reply:
It looks great to me, thanks @rafael-travis 👍